activation/activation.go

```diff
@@ -639,28 +644,38 @@ func (b *Builder) createAtx(ctx context.Context) (*types.ActivationTx, error) {
 	//                           WAITING FOR POET        DEADLINE
 	//                               PROOFS
 	// NiPoST must be ready before start of the next poet round.
-	pubEpoch := b.challenge.PublishEpoch()
+	pubEpoch := challenge.PublishEpoch()
+	poetRoundStart := b.layerClock.LayerToTime((pubEpoch - 1).FirstLayer()).Add(b.poetCfg.PhaseShift)
 	nextPoetRoundStart := b.layerClock.LayerToTime(pubEpoch.FirstLayer()).Add(b.poetCfg.PhaseShift)
 	poetRoundEnd := nextPoetRoundStart.Add(-b.poetCfg.CycleGap)
 	poetProofDeadline := poetRoundEnd.Add(b.poetCfg.GracePeriod)
 
 	b.log.With().Info("building NIPost",
-		log.Stringer("poet round start", nextPoetRoundStart),
-		log.Stringer("pub_epoch", pubEpoch),
-		log.Time("deadline time", poetProofDeadline),
+		log.Time("poet round start", poetRoundStart),
+		log.Time("next poet round start", nextPoetRoundStart),
+		log.Time("poet proof deadline", poetProofDeadline),
+		log.FieldNamed("current epoch", b.currentEpoch()),
+		log.FieldNamed("publish epoch", pubEpoch),
+		log.FieldNamed("target epoch", challenge.TargetEpoch()),
 	)
 
-	challenge := types.PoetChallenge{
-		NIPostChallenge: b.challenge,
+	now := time.Now()
+	if poetRoundStart.Before(now) {
```

Comment:
why not check it right away? the extra info log above seems wasted.

Reply:
IMO, the log above gives a piece of important contextual information for debugging.

Reply:
yeah, but at this point what's there to debug? the challenge is stale... anyhow, not important. just an attempt to reduce logged data.